tests/test_project.py

```diff
@@ -143,6 +143,7 @@ def recursive_update(d, u):
                 project.open_job(dict(a=a, b=b)).init()
                 project.open_job(dict(a=dict(a=a), b=b)).init()
         project._entrypoint = self.entrypoint
+        project._register_aggregates()
```

Comment:
We register jobs after initializing the project. This means that technically, the jobs in the project has changed hence we'll need to re-register aggregates. I think we should make a method called `reregister_aggregates` and use that method instead of using a private method.

Reply:
@vyasr I believe that was the consensus, yes. Create all aggregates on project instantiation, and then offer a public method to update the aggregates but don't do so implicitly.

_However,_ it looks like this method is creating a "mock" project and adding jobs after the project is instantiated, so the normal workflow is subverted. I think the strategy here is acceptable but perhaps it would be better to:
1. Instantiate project and add jobs (this would normally be done in something like an `init.py` script)
2. Create new instance of project (triggers aggregation registration)
3. Alter the new instance's `project._entrypoint` and other "mocking" hacks, then return it

This way any other "constructor" code is called correctly in the future.

Reply:
Yes I would prefer that as well.

Reply:
Please correct me if I'm wrong. I believe the strategy that @bdice proposed would result in changing in the way we instantiate projects in the tests. This should be tackled in a separate pull request. If you guys agree then I'm ready to give this a priority right after we merge this pull request.

Reply:
I think the requested change is relevant enough and small enough that it could be done in this PR. Since all the tests use the `mock_project` method here, isn't it just a matter of adding an extra call to `get_project` or `project_class(...)` before the entrypoint is set? If there's some reason it needs to be more complex, then we can discuss doing it in a follow-up PR.

Reply:
I agree with @vyas, this should be only slightly different (re-instantiate the project instead of calling the private method) and it fits in the scope of this PR.

Reply:
@bdice @vyasr 
I agree with what you suggest but now when I'm thinking more about this, we won't be having any measure to test the behaviour in this pull request because the default aggregate store just iterates over the project and this will always provide us all the jobs present in the project, no matter when we add the job.

I agree that it's just a one-liner change but I'd still want to change this in the other PR (where we also use the `AggrgateStore` class) and remove the `_register_aggregates`  call from the tests for now.

Reply:
I understand your reservation, but whether we call `_register_aggregates` manually or not doesn't change that fact. Right now we're also not really testing whether `_register_aggregates` works for non-default aggregates anyway, so I don't think these two changes are related.